source/MaterialXFormat/File.cpp

```diff
@@ -209,9 +209,20 @@ FilePathVec FilePath::getSubDirectories() const
         while (struct dirent* entry = readdir(dir))
         {
             string path = entry->d_name;
-            if (entry->d_type == DT_DIR && (path != "." && path != ".."))
+            if (path == "." || path == "..") continue;
+
+            auto d_type = entry->d_type;
+            FilePath newDir = *this / path;
+
+            if (d_type == DT_UNKNOWN)
```

Comment:
A filesystem, like XFS, is free to return DT_UNKNOWN if its internal data structure does not store file/dir/link info, forcing an expensive stat call.